packages/sitecore-jss/src/personalization/personalization-decisions-service.ts

```diff
@@ -0,0 +1,175 @@
+}
+
+export type DataFetcherResolver = <T>(config: {
+  /**
+   * The request timeout in milliseconds
+   */
+  timeout?: number;
+}) => HttpDataFetcher<T>;
+
+export type RestPersonalizationDecisionsServiceConfig = {
+  /**
+   * Hostname of decisions service; e.g. http://my.site.core; Default: '', same host as a page
+   */
+  host?: string;
+  /**
+   * Relative path from host to decisions service. Default: /sitecore/api/layout/personalization/decision
+   */
+  route?: string;
+  /**
+   * This value overrides the default service URL.
+   * Note: `host` and `route` options are ignored if `serviceUrl` is set.
+   */
+  serviceUrl?: string;
+  /**
+   * The Sitecore SSC API key your app uses
+   */
+  apiKey: string;
+  /**
+   * The JSS application name
+   */
+  siteName: string;
+  /**
+   * Query string parameters of the current page to pass with request to decision service
+   */
+  currentPageParamsToExtract?: string[];
+  /**
+   * Enables/disables analytics tracking for the Layout Service invocation (default is true).
+   * More than likely, this would be set to false for SSG/hybrid implementations, and the
+   * JSS tracker would instead be used on the client-side: {@link https://jss.sitecore.com/docs/fundamentals/services/tracking}
+   * @default true
+   */
+  tracking?: boolean;
+  /**
+   * The request timeout in milliseconds
+   */
+  timeout?: number;
+  /**
+   * Data fetcher resolver in order to provide custom data fetcher
+   * @see DataFetcherResolver
+   * @see HttpDataFetcher<T>
+   * @see AxiosDataFetcher used by default
+   */
+  dataFetcherResolver?: DataFetcherResolver;
+};
+
+/**
+ * Fetches personalization decisions using the Sitecore REST API.
+ * Uses Axios as the default data fetcher (@see AxiosDataFetcher).
+ */
+export class RestPersonalizationDecisionsService implements PersonalizationDecisionsService {
+  private serviceConfig: RestPersonalizationDecisionsServiceConfig;
+
+  constructor(serviceConfig: RestPersonalizationDecisionsServiceConfig) {
+    this.serviceConfig = {
+      host: '',
+      route: '/sitecore/api/layout/personalization/decision',
+      ...serviceConfig,
+    };
+  }
+
+  /**
+   * Gets personalization decisions.
+   * @param {DecisionsContext} context The decisions context
+   * @returns {Promise<PersonalizationDecisionData>} The personalization decision data
+   */
+  getPersonalizationDecisions(context: DecisionsContext): Promise<PersonalizationDecisionData> {
+    const fetcher = this.serviceConfig.dataFetcherResolver
+      ? this.serviceConfig.dataFetcherResolver<PersonalizationDecisionData>({
+          timeout: this.serviceConfig.timeout,
+        })
+      : this.getDefaultFetcher<PersonalizationDecisionData>();
+    let queryParams = {
+      sc_apikey: this.serviceConfig.apiKey,
+      sc_site: this.serviceConfig.siteName,
+      tracking: this.serviceConfig.tracking ?? true,
+    };
+    if (this.serviceConfig.currentPageParamsToExtract) {
+      queryParams = {
```

Comment:
Object.assign might make this simpler (instead of destructuring an object onto itself)